modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/MacroExpandedComponentAst.java

```diff
@@ -150,9 +171,11 @@ private Object mapErrorMappings(final ComponentParameterAst originalParameter) {
         return originalParameter.getValue()
             .mapRight(mappings -> ((List<ErrorMapping>) mappings)
                 .stream()
-                .map(mapping -> new ErrorMapping(literalsParameters.getOrDefault(mapping.getSource(), mapping.getSource()),
+                .map(mapping -> new ErrorMapping(literalsParameters.getOrDefault(mapping.getSource(), mapping.getSource())
```

Comment:
i don't understand this change. It seems to be altering the logic quite a bit

Reply:
This is necessary to compensate for [this other change](https://github.com/mulesoft/mule/pull/12952/files/701011683b2cff91c81866072b15673267810d8b#r1374944032).
Before, it was stringified earlier, when generating the `propertiesMap` and, immediately, the `literalsParameters` map.
Now we are preserving the `Object` that is computed as the `getValue().getRight()` from the `ComponentParameterAst`. We have a Map<String, Object> instead of a Map<String, String>.
So when we get to this point, I'm doing the `toString` to preserve the previous behavior regarding the output value from this method.